OpenTrons/opentrons-api
# feature(api, robot-server, shared-data): Enable json schema v7 in robot server

# Overview

Enable backend changes for json schema v7. 
closes https://opentrons.atlassian.net/browse/RSS-282.

# Test Plan

- upload a v7 json protocol. (I can supply a few/use fixtures in shared-data)
- make sure analysis passes and that you are able to run the protocol on the robot. 
- smoke test v6 just in case
- there is an integration test as well to test e2e. 


# Changelog

- `file_format_validator` validate json v7.
- `json_file_reader` parse json v7.
- `json_translator` translate v7 commands.
- added `ProtocolSchemaV7` data model in shared data.
- moved shared models between v6 and v7 schema to a shared file.


# Review requests

Changes make sense?
Take a close look at Params changes for v7. are we missing something?
are there test cases I should add? 

# Risk assessment

low. moved v6 shared models to a shared file. added v7 infrastructure - need to smoke test both v6, v7 json protocols
